File: primes.lua
function prime(x)
    for i = 2,x/2,1
    do 
        if x % i == 0 then
            return false
        end
    end
    return true
end
for i = 2,100,1
do
    if prime(i) then
        print(i)
    end
end

File: primes.zig
const std = @import("std");
pub fn prime(x : i32) bool {
    var y: i32 = @divFloor(x,2);
    while(y > 1) : (y -= 1)
        if(@mod(x,y) == 0)
            return false;
    return true;
}
pub fn main() !void {   
    var i: i32 = 2;
    while (i < 100) : (i += 1)
        if(prime(i))
            try std.io.getStdOut().writer().print("{}\n", .{i});
}